torch/optim/lr_scheduler.py

```diff
@@ -82,12 +121,23 @@ def step(self, epoch=None):
                               "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
         self._step_count += 1
 
+
         if epoch is None:
-            epoch = self.last_epoch + 1
-        self.last_epoch = epoch
-        for param_group, lr in zip(self.optimizer.param_groups, self.get_lr()):
+            self.last_epoch = self.last_epoch + 1
+            values = self._compute_values()
+        else:
+            warnings.warn(EPOCH_DEPRECATION_WARNING, DeprecationWarning)
+            self.last_epoch = epoch
+            try:
+                values = self._compute_values_closed_form()
+            except NotImplementedError:
+                values = self._compute_values()
```

Comment:
We are not ignoring the epoch, see the line before `try`: `self.last_epoch = epoch`.

We could show the warning only when actually using the closed form, since some schedulers didn't have a closed form in the first place. Thoughts?